packages/lib-utils/src/app/AppInitSDK.tsx

```diff
@@ -0,0 +1,51 @@
+import { consoleLogger } from '@monorepo/common';
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import type { UtilsConfig } from '../config';
+import { setUtilsConfig } from '../config';
+import type { InitAPIDiscovery } from '../types/api-discovery';
+import { initAPIDiscovery } from './api-discovery';
+import { useReduxStore } from './redux';
+
+type AppInitSDKProps = React.PropsWithChildren<{
```

Comment:
I disagree with @vojtechszocs on this one. `children` is required -- but since `undefined` is part of `React.ReactNode` we'll need to use `ReactElement`. It's important to note this component is not intended to be used next to your app... it needs to be wrapping your app so we can provide all things important for hooks and usages inside your app.

Please change to:

```
type AppInitSDKProps = {
  children: React.ReactElement; // this is only a single JSX component (their app)
  configurations: {
    apiDiscovery?: InitAPIDiscovery;
    appFetch: UtilsConfig['appFetch'];
  };
};
```